src/actions/guides/database/querying.cr

```diff
@@ -7,294 +7,240 @@ class Guides::Database::Querying < GuideAction
+    > However, for brevity, we will use `COLUMNS`.
 
-    Lucky will accept the type defined in the model (e.g. `Int32` for `age`) or a
-    `String`. If you pass it a `String` then Lucky will attempt to parse it and use
-    the type defined for the column.
+    ### Select all
 
-    For example:
+    `SELECT COLUMNS FROM users`
 
     ```crystal
-    # Lucky parses the string and use an Int32 in the query
-    UserQuery.new.age("30")
-
-    # Won't accept other types
-    # Fails at compile time because you can't pass a Float64
-    UserQuery.new.age(30.0)
+    users = UserQuery.all
     ```
 
-    > Note that this will change in the future but this is the current behavior. In
-    the future only the defined type (`Int32` in this example) will work. You will
-    then need to manually parse it. Don't worry though, Lucky will add some helpers
-    to actions and the model to make this easy and maintainable.
-
-    ## Ordering results
-
-    ````crystal
-    UserQuery.new.age.desc_order
-    # or
-    UserQuery.new.age.asc_order
-    ````
+    ### Select one
 
-    ## Chaining methods
-
-    You can chain where clauses and most other query methods.
+    Selecting the first user `SELECT COLUMNS FROM users LIMIT 1`
 
     ```crystal
-    UserQuery.new.age.gt(28).age.desc_order
+    user = UserQuery.first
     ```
 
-    ## Running the query
-
-    The query does not actually hit the database until a method is called to fetch a
-    result or iterate over results.
-
-    Some of the most common methods are:
+    Selecting the last user `SELECT COLUMNS FROM users ORDER BY users.id DESC LIMIT 1`
 
-    * `first`
-    * `find`
-    * `each`
+    ```crystal
+    user = UserQuery.last
+    ```
 
-    For example:
+    Selecting the user with `id = 3` `SELECT COLUMNS FROM users WHERE id = 3 LIMIT 1`
 
     ```crystal
-    # The query is not yet run
-    query = UserQuery.new
-    query.name("Sally")
-    query.age(30)
-
-    # The query will run once `each` is called
-    # Results are not cached so a request will be made every time you call `each`
-    query.each do |user|
-      pp user.name
-    end
+    user = UserQuery.find(3)
     ```
 
-    ## Finding by ID
+    ### Select distinct / distinct on
 
-    ```crystal
-    id = 1
-    UserQuery.new.find(id)
+    `SELECT DISTINCT COLUMNS FROM users`
 
-    # You can also chain methods before calling find, first, etc.
-    UserQuery.new.age(30).find(1)
+    ```crystal
+    UserQuery.new.distinct
     ```
 
-    ## Ensuring no results are returned
+    Select distinct rows based on the `name` column `SELECT DISTINCT ON (users.name) FROM users`
 
     ```crystal
-    UserQuery.new.none
+    UserQuery.new.distinct_on(&.name)
     ```
 
-    This can be helpful when under certain conditions you want the resultd to be empty.
+    > Notice these two methods are called in the instance of `UserQuery`, and not the class.
```

Comment:
This isn't *quite* right actually all the methods are instance methods and maybe that's what should be documented here. The `first/last/etc` on the class are shortcuts that call `new.first/last/etc`. So I'd suggest we use those and add a separate section `###` that says `Select shortcut` methods.

May also be worth mentioning that you can chain these like `UserQuery.new.age(30).first`